app/src/main/java/ru/yandex/practicum/contacts/presentation/sort/SortTypeUI.java

```diff
@@ -22,6 +23,12 @@ public boolean isSelected() {
         return selected;
     }
 
+    @Override
+    public boolean theSameAs(SortTypeUI other){
+        if (other == null) return false;
+        return this.sortType.equals(other.sortType);
```

Comment:
Аналогично `FilterContactType` тип `SortType` является енамом и его можно смело сравнивать через `==`:
```suggestion
        return this.sortType == other.sortType;
```